src/SwitchField/SwitchField.test.jsx

```diff
@@ -18,6 +18,19 @@ describe("SwitchField", () => {
     await user.click(screen.getByRole("checkbox"));
     expect(onChange).toHaveReturnedWith(expected);
   });
+  it("can be uncontrolled and have a defaultChecked prop", async () => {
+    const user = userEvent.setup();
+    const onChange = vi.fn(event => event.target.checked);
+    render(
+      <SwitchField
+        defaultChecked={true}
+        options={["Option A", "Option B"]}
+        onChange={onChange}
+      />
+    );
+    await user.click(screen.getByRole("checkbox"));
```

Comment:
Your tests seem to either check the rendered value given a defaultValue, or in this case check the onChange handler after a user interaction. I suggest we need both for all the touched components in this PR.

Reply:
In this instance I was only checking the onChange to infer the default value but have updated it to just check the default value like the checkbox. The onChange events for each component haven't been touched and will just reflect the change event  as per existing behaviour.